zetta_utils/augmentations/tensor.py

```diff
@@ -169,3 +171,114 @@ def square_tile_pattern_aug(  # pylint: disable=too-many-locals
     if len(data.shape) == 3:
         result = result.squeeze(-1)
     return result
+
+
+@builder.register("apply_to_random_sections")
+@prob_aug
+@typechecked
+def apply_to_random_sections(
+    data: TensorTypeVar,
+    fn: Callable[[TensorTypeVar], TensorTypeVar],
+    num_sections: Union[distributions.Distribution, int] = 1,
+) -> TensorTypeVar:
+    assert data.ndim == 4, "Input must be CXYZ tensor"
+    num_sections_chosen = int(distributions.to_distribution(num_sections)())
+    chosen_sections = random.sample(range(0, data.shape[-1]), num_sections_chosen)
+    result = clone(data)
+    for i in chosen_sections:
+        processed = astype(fn(data[..., i]), result)
+        assert isinstance(processed, type(result))
+        result[..., i] = processed  # type: ignore # mypy bug
+    return result
+
+
+@overload
+def apply_to_random_boxes(
+    data: TensorTypeVar,
+    fn: Callable[[TensorTypeVar], TensorTypeVar],
+    box_size: Union[distributions.Distribution, float],
+    num_boxes: Union[distributions.Distribution, int] = ...,
+    density: None = ...,
+    vary_box_sizes: bool = ...,
+    allow_partial_boxes: bool = ...,
+) -> TensorTypeVar:
+    ...
+
+
+@overload
+def apply_to_random_boxes(
+    data: TensorTypeVar,
+    fn: Callable[[TensorTypeVar], TensorTypeVar],
+    box_size: Union[distributions.Distribution, float],
+    num_boxes: None = ...,
+    density: Union[distributions.Distribution, float] = ...,
+    vary_box_sizes: bool = ...,
+    allow_partial_boxes: bool = ...,
+) -> TensorTypeVar:
+    ...
+
+
+@builder.register("apply_to_random_boxes")
+@prob_aug
+@typechecked
+def apply_to_random_boxes(
+    data,
+    fn,
+    box_size,
+    num_boxes=1,
```

Comment:
I'm actually uncertain about the practical usefulness of specifying the (distribution of the) number of boxes. Are there any use cases that aren't addressed by specifying the density?

Reply:
It could be useful. For example let's say we just want to occlude one single box. Feels like a desirable functionality and it's already implemented, so no reason to remove it